examples/object_detection_2d/object_detection_2d/extended/seed_test_suite.py

```diff
@@ -0,0 +1,208 @@
+from object_detection_2d.extended.workflow import TestSuite
+
+import kolena
+from kolena.workflow.annotation import BoundingBox
+
+
+def load_transportation_data() -> Dict[str, List[ExtendedBoundingBox]]:
+    s3 = s3fs.S3FileSystem()
+
+    try:
+        with s3.open(S3_ANNOTATION_FILE_PATH, "r") as file:
+            coco_data = json.load(file)
+    except OSError as e:
+        print(e, "\nPlease ensure you have set up AWS credentials.")
+        exit()
+
+    # gather image IDs with the Attribution 2.0 license - https://creativecommons.org/licenses/by/2.0/
+    ids = {int(entry["id"]) for entry in coco_data["images"] if entry["license"] == 4}
+
+    # class id to string label
+    label_map: Dict[int, str] = {int(category["id"]): category["name"] for category in coco_data["categories"]}
+
+    # cache bounding boxes per image
+    image_to_boxes: Dict[int, List[ExtendedBoundingBox]] = defaultdict(lambda: [])
+    for annotation in coco_data["annotations"]:
+        image_id = annotation["image_id"]
+        label = label_map[annotation["category_id"]]
+
+        # check that the box is in a valid image, not a crowd box, and in the transportation supercategory
+        if image_id in ids and not annotation["iscrowd"] and label in TRANSPORTATION_LABELS:
+            bbox = annotation["bbox"]
+            top_left = (bbox[0], bbox[1])
+            bottom_right = (bbox[0] + bbox[2], bbox[1] + bbox[3])
+            bounding_box = ExtendedBoundingBox(top_left, bottom_right, label, occluded=False)
+            image_to_boxes["COCO_val2014_" + str(image_id).zfill(12) + ".jpg"].append(bounding_box)
+    return image_to_boxes
+
+
+def create_complete_transportation_case(args: Namespace) -> TestCase:
+    print("loading S3 files...")
+    image_to_boxes = load_transportation_data()
+    data = pd.read_csv(args.metadata)
+
+    # create a test sample object and a ground truth object per image
+    test_samples_and_ground_truths: List[Tuple[TestSample, GroundTruth]] = []
+    for _, row in data.iterrows():
+        image_name = row["name"]
+        test_sample = TestSample(
+            locator=S3_IMAGE_LOCATION + image_name,
+            metadata={
+                "image_width": row["width"],
+                "image_height": row["height"],
+                "brightness": row["brightness"],
+            },
+        )
+
+        ground_truth = GroundTruth(
+            bboxes=image_to_boxes[image_name],
+            ignored_bboxes=[],
+        )
+
+        test_samples_and_ground_truths.append((test_sample, ground_truth))
+
+    # create the complete test case, not attached to any test suite
+    complete_test_case = TestCase(
+        f"complete transportation :: {DATASET} [{WORKFLOW}]",
+        test_samples=test_samples_and_ground_truths,
+        reset=True,
+    )
+
+    return complete_test_case
+
+
+def seed_test_suite_by_brightness(test_suite_name: str, complete_test_case: TestCase) -> None:
```

Comment:
Since this example is just meant to illustrate the extension of the workflow, we can just make a single test case test suite and pare this `seed_test_suite` script down significantly. Your call

Reply:
I agree that it will pare it down quite a bit but I prefer to leave it as is because `seed_test_run` has a list of `TestSuite` names specified by default so you can run the script without passing in the test suite name as an argument but the default argument won't work if I make this change.